### PR TITLE
upgrade sample of kafka operator to 0.31.x

### PR DESCRIPTION
**Pull request by yanmxa**
Signed-off-by: myan <myan@redhat.com>
Reference: https://github.com/stolostron/backlog/issues/26674

### CODE REVIEW

File: operator/config/samples/transport/deploy_kafka.sh [hunk 1]
```diff
@@ -13,11 +13,10 @@ fi
 
 # install community kafka operator
 kubectl create namespace kafka --dry-run=client -o yaml | kubectl apply -f -
-KAFKA_OPERATOR=${KAFKA_OPERATOR:-"strimzi-cluster-operator-v0.23.0"}
 kubectl apply -f ${currentDir}/kafka-operator.yaml
 
 # wait until operator is ready
-operatorDeployed=$(kubectl -n kafka get Deployment/$KAFKA_OPERATOR --ignore-not-found)
+operatorDeployed=$(kubectl pods -n kafka -l name=strimzi-cluster-operator --ignore-not-found | grep Running || true)
```
Comment (clyang82) [inline operator/config/samples/transport/deploy_kafka.sh:19]: missed `get`?

**Comment by clyang82** [comment]
FYI @bjoydeep have updated to the latest version for kafka.

**Comment by clyang82** [comment]
@yanmxa from e2e failure message, kafka cluster cannot become available.

**Comment by yanmxa** [comment]
> @yanmxa from e2e failure message, kafka cluster cannot become available.

Yes, It was caused by using Kafka version too low and the Kafka CR incompatibility. It should work now.

**Review by morvencao — COMMENTED** [review]
/lgtm

**Pull request merged**